opennms-dao/src/main/java/org/opennms/netmgt/dao/hibernate/SnmpInterfaceDaoHibernate.java

```diff
@@ -53,6 +53,14 @@ public OnmsSnmpInterface findByNodeIdAndIfIndex(Integer nodeId, Integer ifIndex)
         
     }
 
+    @Override
+    public List<OnmsSnmpInterface> findByNodeId(Integer nodeId) {
+        Assert.notNull(nodeId, "nodeId may not be null");
+        return find("select distinct snmpIf from OnmsSnmpInterface as snmpIf where snmpIf.node.id = ?",
```

Comment:
Does this really need to be distinct? As this includes the ID, distinctiveness should be inherent, right?

Reply:
I am not familiar enough with the underlying relational model. AFAIU the same snmp interface can be referenced by several ip interfaces, a node in turn may have several ip interfaces. In summary, it seems that the same snmp interfaces can be related to a node via different ip interfaces.

Reply:
This is not true. The `SnmpInterface` is directly associated to a `Node` in a 1:n relation.

Reply:
You are right, `distinct` is not necessary. I was mislead by other selects in that class that include `distinct`. I removed these occurrences, too.